### PR TITLE
NIAD-1065: update interchange header sender and recipient examples to life-like 15 digit NHS codes

### PR DESCRIPTION
**Pull request by n-r-b**
## Description

Edifact interchange header `recipient` and `sender` examples have been updated to life-like 15 digit NHS codes throughout the adaptor, as we have been using unrealistic codes copied from NHAIS. 

I've used the recipient/sender examples mentioned in the README and edifact files.

## Jira Ticket

https://gpitbjss.atlassian.net/browse/NIAD-1065

**AC3 only**

## Checklist

These are items (excluding GitHub Checks) which should be confirmed before a branch is ready to merge.

- [x] Acceptance Criteria met
- [x] Commit messages are meaningful
- [x] Manually tested
- [x] Self-reviewed your code
- [ ] Code reviewed from two other developers
- [x] If your pull request depends on any other, please link them in the description
- [ ] main branch is passing/green on CI
- [x] Add/update any relevant documentation

### CODE REVIEW

File: src/intTest/java/uk/nhs/digital/nhsconnect/lab/results/mesh/MeshClientIntegrationTest.java [hunk 1]
```diff
@@ -87,7 +87,7 @@ void when_callingGetMessageWithLargeContentAndWrongWorkflowId_expect_meshWorkflo
 
     @SneakyThrows
     private MeshMessageId sendLargeMessageWithWrongWorkflowId() {
-        OutboundMeshMessage messageForMappingMailboxId = new MeshMessage().setRecipient("XX11");
+        OutboundMeshMessage messageForMappingMailboxId = new MeshMessage().setRecipient("000000004400001");
```
Comment (n-r-b) [inline src/intTest/java/uk/nhs/digital/nhsconnect/lab/results/mesh/MeshClientIntegrationTest.java:90]:
```suggestion
        OutboundMeshMessage messageForMappingMailboxId = new MeshMessage().setRecipient(RECIPIENT);
```

Comment (n-r-b) [reply src/intTest/java/uk/nhs/digital/nhsconnect/lab/results/mesh/MeshClientIntegrationTest.java]: Done

File: src/intTest/resources/edifact/multi_pathology.edifact.dat [hunk 2]
```diff
@@ -1,4 +1,4 @@
-UNB+UNOC:3+XX11+000000024600002:80+100301:1751+1015++MEDRPT++1'
+UNB+UNOC:3+000000004400001+000000024600002:80+100301:1751+1015++MEDRPT++1'
```
Comment (graham-andrew) [inline src/intTest/resources/edifact/multi_pathology.edifact.dat:1]: Here and in other examples:

Do we need the :80 after the 15 digit codes? The edifact word document seems to suggest so

Comment (n-r-b) [reply src/intTest/resources/edifact/multi_pathology.edifact.dat:1]: Yes we do, but we'll need to make a change to the interchange header parser (i think). Outside of this PR, i still have a couple of things to do before this ticket is complete,  so I'll create a separate PR for the `:80` part.

**Review by graham-andrew — APPROVED** [review]
Looks good to me, as long as that change regarding the :80 is done in another PR

**Review by andrewwa-kainos — APPROVED** [review]
LGTM

**Pull request merged**